### PR TITLE
Adjust to recent api changes

### PR DESCRIPTION
**Pull request by OleMchls**
also improves error handling

PTAL @aeden

### CODE REVIEW

File: config.example.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 var config = {}
+config.session = {}
```
Comment (aeden) [inline config.example.js:4]: Why was this added?

**Review by aeden — APPROVED** [review]
Just one quick question, otherwise it's fine to merge.

File: config.example.js (hunk 1 above)
Comment (OleMchls) [reply config.example.js:4]: Because you can't set a property on undefined:

```node
👌  hello-domains-node git:(update--to-recent-api-changes) node
> var config = {}
undefined
> config.session.secret = 'somelongrandomstring';
TypeError: Cannot set property 'secret' of undefined
    at repl:1:23
    at realRunInThisContextScript (vm.js:22:35)
    at sigintHandlersWrap (vm.js:98:12)
    at ContextifyScript.Script.runInThisContext (vm.js:24:12)
    at REPLServer.defaultEval (repl.js:346:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.onLine (repl.js:545:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:188:7)
>
```

Comment (aeden) [reply config.example.js:4]: The session secret is not set from within the code, it's set in https://github.com/dnsimple/hello-domains-node/blob/master/config.example.js - does the config loader automatically handle creation of nested objects then? Was the example breaking before you added this?

Comment (OleMchls) [reply config.example.js:4]: You pointed to the file that contained the change 😕 It was breaking for me. IDK maybe a node version thing?

Comment (aeden) [reply config.example.js:4]: OK, go ahead and 🚢  it.
